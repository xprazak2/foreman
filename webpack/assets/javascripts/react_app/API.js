import axios from 'axios';
import qs from 'qs';

const getcsrfToken = () => {
  const token = document.querySelector('meta[name="csrf-token"]');

  return token ? token.content : '';
};

axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
axios.defaults.headers.common['X-CSRF-Token'] = getcsrfToken();

export default {
  get(url, headers = {}, params = {}) {
    console.log(params)
    return axios.get(url, {
      headers,
      params,
      paramsSerializer: parameters => qs.stringify(parameters)
    });
  },
  put(url, data = {}, headers = {}) {
    return axios.put(
      url, data,
      {
        headers,
      },
    );
  },
  post(url, data = {}, headers = {}) {
    return axios.post(
      url, data,
      {
        headers,
      },
    );
  },
  delete(url, headers = {}) {
    return axios.delete(
      url,
      {
        headers,
      },
    );
  },
  patch(url, data = {}, headers = {}) {
    return axios.patch(
      url, data,
      {
        headers,
      },
    );
  },
};
