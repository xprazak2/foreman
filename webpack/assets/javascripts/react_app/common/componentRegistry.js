import React from 'react';
import store from '../redux';
import PieChart from '../components/common/charts/PieChart/';
import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
import PowerStatus from '../components/hosts/powerStatus/';
import NotificationContainer from '../components/notifications/';
import ToastsList from '../components/toastNotifications/';
import StorageContainer from '../components/hosts/storage/vmware/';
import _ from 'lodash';

let componentRegistry = () => {
    let registry = {
      PieChart: {
        type: PieChart,
        store: true,
        data: true
      },
      StatisticsChartsList: {
        type: StatisticsChartsList,
        store: true,
        data: true
      },
      PowerStatus: {
        type: PowerStatus,
        store: true,
        data: true
      },
      NotificationContainer: {
        type: NotificationContainer,
        store: true,
        data: true
      },
      ToastNotifications: {
        type: ToastsList,
        store: true,
        data: false
      }
    };

    return function () {
      let register = (name, component) => {
        if (registry[name]) {
          throw 'Component name already taken';
        }
        registry[name] = component;
        console.log('Component registered with name ' + name);
        return registry;
      };

      let getComponent = (name) => {
        return registry[name];
      };

      let registeredComponents = () => {
        return _.map(registry, (value, key) => {
          return key;
        }).join(', ');
      };

      let markup = (name, data, store) => {
        let currentComponent = getComponent(name);
        if (!currentComponent) {
          throw `Component not found:  ${name} among ${registeredComponents()}`
        }
        let ComponentName = currentComponent.type;
        return <ComponentName data={ currentComponent.data ? data : undefined } store={ currentComponent.store ? store : undefined } />;
      };

      return { register, getComponent, markup };
    }();
};

export default componentRegistry();
