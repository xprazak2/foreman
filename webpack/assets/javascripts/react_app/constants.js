// API
export const STATUS = {
  PENDING: 'PENDING',
  RESOLVED: 'RESOLVED',
  ERROR: 'ERROR',
};

// Documentation
export const FOREMAN_VERSION = '1.23';
export const DOCUMENTATION_URL = `https://theforeman.org/manuals/${FOREMAN_VERSION}/index.html`;

// Pagination
export const PAGE_OPTIONS = [5, 10, 25, 50];

// Search
export const getControllerSearchProps = (
  controller,
  id = 'searchBar',
  canCreate = true
) => ({
  controller,
  autocomplete: {
    id,
    searchQuery: '',
    url: `${controller}/auto_complete_search`,
    useKeyShortcuts: true,
  },
  bookmarks: {
    url: '/api/bookmarks',
    canCreate,
    documentationUrl: `${DOCUMENTATION_URL}#4.1.5Searching`,
  },
});
