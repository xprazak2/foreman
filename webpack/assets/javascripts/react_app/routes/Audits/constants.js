import {
  DOCUMENTATION_URL,
  PAGE_OPTIONS,
  getControllerSearchProps,
} from '../../constants';

export const AUDITS_PAGE_FETCH = 'AUDITS_PAGE_FETCH';
export const AUDITS_PAGE_CHANGE_PARAMS = 'AUDITS_PAGE_CHANGE_PARAMS';
export const AUDITS_PAGE_SHOW_MESSAGE = 'AUDITS_PAGE_SHOW_MESSAGE';
export const AUDITS_PAGE_HIDE_MESSAGE = 'AUDITS_PAGE_HIDE_MESSAGE';
export const AUDITS_PAGE_NEXT_PENDING = 'AUDITS_PAGE_NEXT_PENDING';
export const AUDITS_PAGE_NEXT_RESOLVED = 'AUDITS_PAGE_NEXT_RESOLVED';
export const AUDITS_PAGE_PREV_PENDING = 'AUDITS_PAGE_PREV_PENDING';
export const AUDITS_PAGE_PREV_RESOLVED = 'AUDITS_PAGE_PREV_RESOLVED';
export const AUDITS_PAGE_CLEAR_CACHE = 'AUDITS_PAGE_CLEAR_CACHE';

export const AUDITS_CURRENT = 'audits';
export const AUDITS_NEXT = 'nextAudits';
export const AUDITS_PREV = 'prevAudits';
export const AUDITS_PATH = '/audits';
export const AUDITS_API_PATH = `${AUDITS_PATH}/load`;

export const AUDITS_PER_PAGE_OPTIONS = PAGE_OPTIONS;
export const AUDITS_DOC_URL = `${DOCUMENTATION_URL}#4.1.4Auditing`;
export const AUDITS_SEARCH_PROPS = getControllerSearchProps('audits');
