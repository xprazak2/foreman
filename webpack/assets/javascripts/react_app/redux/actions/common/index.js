import API from '../../../API';

export const ajaxRequestAction = ({
  dispatch,
  requestAction,
  successAction,
  failedAction,
  url,
  item,
  headers = {},
  params = {},
}) => {
  dispatch({ type: requestAction, payload: item });
  return API.get(url, headers, params)
    .then(({ data }) => dispatch({ type: successAction, payload: { ...item, ...data } }))
    .catch(error => dispatch({ type: failedAction, payload: { error, item } }));
};
