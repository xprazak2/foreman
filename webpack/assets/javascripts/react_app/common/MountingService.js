import React from 'react';

// import StorageContainer from '../components/hosts/storage/vmware/';
// import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
// import PowerStatus from '../components/hosts/powerStatus/';
// import NotificationContainer from '../components/notifications/';
// import ToastsList from '../components/toastNotifications/';
// import PieChart from '../components/common/charts/PieChart/';
import ReactDOM from 'react-dom';
import store from '../redux';
import componentRegistry from './componentRegistry';

export function mount(component, selector, data) {
  // const components = {
  //   PieChart: {
  //     type: PieChart,
  //     markup: <PieChart data={ data } />
  //   },
  //   StatisticsChartsList: {
  //     type: StatisticsChartsList,
  //     markup: <StatisticsChartsList store={store} data={data}/>
  //   },
  //   PowerStatus: {
  //     type: PowerStatus,
  //     markup: <PowerStatus store={store} data={data}/>
  //   },
  //   NotificationContainer: {
  //     type: NotificationContainer,
  //     markup: <NotificationContainer store={store} data={data} />
  //   },
  //   ToastNotifications: {
  //     type: ToastsList,
  //     markup: <ToastsList store={store} />
  //   },
  //   StorageContainer: {
  //     type: StorageContainer,
  //     markup: <StorageContainer store={store} data={data} />
  //   }
  // };
  const reactNode = document.querySelector(selector);

  if (reactNode) {
    ReactDOM.unmountComponentAtNode(reactNode);
    ReactDOM.render(componentRegistry.markup(component, data, store), reactNode);
  } else {
    const componentName = componentRegistry(component).type.name;

    // eslint-disable-next-line no-console
    console.log(`Cannot find \'${selector}\' element for mounting the \'${componentName}\'`);
  }
}
