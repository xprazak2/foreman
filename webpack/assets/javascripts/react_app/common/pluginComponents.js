import componentRegistry from './componentRegistry';
import _ from 'lodash';

export default function register(bundleNames) {
  console.log(bundleNames)
  _.forEach(bundleNames, (name) => {
    let pluginComponents = require(name + '/components');
    registerPluginComponents(pluginComponents);
  });
}

let registerPluginComponents = (pluginComponents) => {
  for (let [name, component] of Object.entries(pluginComponents)) {
    componentRegistry.register(name, component);
  }
};
