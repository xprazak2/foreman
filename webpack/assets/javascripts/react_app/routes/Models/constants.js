import { getManualURL } from '../../common/helpers';
import { getControllerSearchProps } from '../../constants';

export const MODELS_PAGE_DATA_RESOLVED = 'MODELS_PAGE_DATA_RESOLVED';
export const MODELS_PAGE_DATA_FAILED = 'MODELS_PAGE_DATA_FAILED';
export const MODELS_PAGE_HIDE_LOADING = 'MODELS_PAGE_HIDE_LOADING';
export const MODELS_PAGE_SHOW_LOADING = 'MODELS_PAGE_SHOW_LOADING';
export const MODELS_PAGE_UPDATE_QUERY = 'MODELS_PAGE_UPDATE_QUERY';
export const MODELS_PAGE_CLEAR_ERROR = 'MODELS_PAGE_CLEAR_ERROR';

export const MODELS_SEARCH_PROPS = getControllerSearchProps('models');
