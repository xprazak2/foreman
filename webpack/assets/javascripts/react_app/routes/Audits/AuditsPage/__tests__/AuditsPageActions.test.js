import API from '../../../../API';

import { testActionSnapshotWithFixtures } from '../../../../common/testHelpers';
import {
  fetchAudits,
  fetchAndPush,
  initializeAudits,
  auditSearch,
  changePage,
  changePerPage,
  displayMessage,
  fetchingNext,
  fetchingPrev,
  resolvedNext,
  resolvedPrev,
  clearCache,
} from '../AuditsPageActions';
import { AUDITS_NEXT, AUDITS_PREV } from '../../constants';
import {
  getMock,
  responseMock,
  emptyResponseMock,
} from '../AuditsPage.fixtures';

jest.mock('../../../../API');

const runWithGetState = (state, action, ...params) => dispatch => {
  const getState = () => ({
    auditsPage: state,
  });
  action(...params)(dispatch, getState);
};

const runFetchAuditsAPI = (state, resourceMock, serverMock) => {
  API.get.mockImplementation(serverMock);

  return runWithGetState(state, fetchAudits, resourceMock);
};

const fixtures = {
  'should fetch Audits': () =>
    runFetchAuditsAPI(
      { showMessage: false },
      getMock,
      async () => responseMock
    ),

  'should fetch nextAudits': () =>
    runFetchAuditsAPI(
      { showMessage: false },
      {
        ...getMock,
        view: AUDITS_NEXT,
      },
      async () => responseMock
    ),

  'should fetch prevAudits': () =>
    runFetchAuditsAPI(
      { showMessage: false },
      {
        ...getMock,
        view: AUDITS_PREV,
      },
      async () => responseMock
    ),

  'should fetch empty Audits': () =>
    runFetchAuditsAPI(
      { showMessage: false },
      { ...getMock, searchQuery: 'no-such-audit' },
      async () => emptyResponseMock
    ),

  'should fetch Audits and remove emptyState': () =>
    runFetchAuditsAPI({ showMessage: true }, getMock, async () => responseMock),

  'should fetch Audits and fail': () =>
    runFetchAuditsAPI({ showMessage: true }, getMock, async () => {
      throw new Error('some-error');
    }),

  'should changePage': () =>
    runWithGetState(
      {
        perPage: 20,
        searchQuery: 'search',
      },
      changePage,
      5
    ),

  'should changePage and fetchNext': () =>
    runWithGetState(
      {
        page: 3,
        perPage: 20,
        searchQuery: 'search',
        nextAudits: [1, 2],
      },
      changePage,
      4
    ),

  'should changePage and fetchPrev': () =>
    runWithGetState(
      {
        page: 3,
        perPage: 20,
        searchQuery: 'search',
        prevAudits: [1, 2],
      },
      changePage,
      2
    ),

  'should changePerPage': () =>
    runWithGetState(
      {
        perPage: 5,
        itemCount: 20,
        searchQuery: 'search',
      },
      changePerPage,
      25
    ),

  'should changePerPage without API': () =>
    runWithGetState(
      {
        audits: [1, 2, 3, 4, 5, 6, 7],
        searchQuery: 'search',
        perPage: 25,
      },
      changePerPage,
      5
    ),

  'should auditSearch': () =>
    runWithGetState(
      {
        perPage: 5,
      },
      auditSearch,
      'search'
    ),

  'should fetchAndPush': () => fetchAndPush(getMock),

  'should initializeAudits w/replace': () =>
    runWithGetState(
      { searchQuery: 'search' },
      initializeAudits,
      { page: 2, perPage: 20, searchQuery: 'search' },
      true
    ),
  'should initializeAudits': () =>
    runWithGetState({ searchQuery: 'search' }, initializeAudits, {
      page: 2,
      perPage: 20,
      searchQuery: 'search',
    }),
  'should displayMessage': () => displayMessage('text', 'error'),
  'should fetchingNext': () => fetchingNext(),
  'should fetchingPrev': () => fetchingPrev(),
  'should resolvedNext': () => resolvedNext(),
  'should resolvedPrev': () => resolvedPrev(),
  'should clearCache': () => clearCache(),
};

describe('AuditsPage actions', () => testActionSnapshotWithFixtures(fixtures));
