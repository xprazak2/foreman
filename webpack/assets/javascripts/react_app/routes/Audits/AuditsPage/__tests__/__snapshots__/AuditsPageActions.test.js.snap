// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditsPage actions should auditSearch 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    [Function],
  ],
]
`;

exports[`AuditsPage actions should changePage 1`] = `
Array [
  Array [
    Object {
      "type": "AUDITS_PAGE_CLEAR_CACHE",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    [Function],
  ],
]
`;

exports[`AuditsPage actions should changePage and fetchNext 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "audits": Array [
          1,
          2,
        ],
        "nextAudits": Array [],
        "page": 4,
        "perPage": 20,
        "prevAudits": undefined,
        "searchQuery": "search",
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`AuditsPage actions should changePage and fetchPrev 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "audits": Array [
          1,
          2,
        ],
        "nextAudits": undefined,
        "page": 2,
        "perPage": 20,
        "prevAudits": Array [],
        "searchQuery": "search",
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`AuditsPage actions should changePerPage 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    [Function],
  ],
]
`;

exports[`AuditsPage actions should changePerPage without API 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "audits": Array [
          1,
          2,
          3,
          4,
          5,
        ],
        "perPage": 5,
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
]
`;

exports[`AuditsPage actions should clearCache 1`] = `
Object {
  "type": "AUDITS_PAGE_CLEAR_CACHE",
}
`;

exports[`AuditsPage actions should displayMessage 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "text": "text",
          "type": "error",
        },
        "showMessage": true,
      },
      "type": "AUDITS_PAGE_SHOW_MESSAGE",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch Audits 1`] = `
Array [
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "audits": Array [
          Object {
            "action": "update",
            "action_display_name": "updated",
            "affected_locations": Array [
              Object {
                "name": "test_loc1",
                "url": "/locations/2-test_loc1/edit",
              },
              Object {
                "name": "test_loc2",
                "url": "/locations/6-test_loc2/edit",
              },
              Object {
                "name": "test_loc3",
                "url": "/locations/9-test_loc3/edit",
              },
            ],
            "affected_organizations": Array [
              Object {
                "name": "test_org1",
                "url": "/organizations/1-test_org1/edit",
              },
              Object {
                "name": "test_org2",
                "url": "/organizations/3-test_org2/edit",
              },
              Object {
                "name": "test_org3",
                "url": "/organizations/5-test_org3/edit",
              },
            ],
            "allowed_actions": Array [
              Object {
                "css_class": "btn btn-default",
                "title": "Host details",
                "url": "/hosts/host-foo.example.com",
              },
            ],
            "associated_id": null,
            "associated_name": null,
            "associated_type": null,
            "audit_title": "host-foo.example.com",
            "audit_title_url": "/audits?search=type+%3D+host+and+auditable_id+%3D+9",
            "auditable_id": 9,
            "auditable_name": "host-foo.example.com",
            "auditable_type": "Host::Base",
            "audited_changes": Object {
              "comment": Array [
                "",
                "This is info about host for audit",
              ],
              "root_pass": Array [
                "[redacted]",
                "[redacted]",
              ],
            },
            "audited_changes_with_id_to_label": Array [
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[redacted]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "[redacted]",
                  },
                ],
                "name": "Root pass",
              },
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[empty]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "This is info about host for audit",
                  },
                ],
                "name": "Comment",
              },
            ],
            "audited_type_name": "Host",
            "comment": null,
            "created_at": "2018-08-13 00:34:55 -1100",
            "id": 234,
            "remote_address": "127.0.0.1",
            "request_uuid": "c134239d-8ac3-494b-9962-35133fe153ba",
            "user_id": 4,
            "user_info": Object {
              "audit_path": "/audits?search=id+%3D+234",
              "display_name": "Admin ",
              "login": "admin",
              "search_path": "/audits?search=user+%3D+admin",
            },
            "user_type": null,
            "username": "Admin User",
            "version": 2,
          },
        ],
        "itemCount": 1,
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch Audits and fail 1`] = `
Array [
  Array [
    Object {
      "type": "AUDITS_PAGE_HIDE_MESSAGE",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch Audits and remove emptyState 1`] = `
Array [
  Array [
    Object {
      "type": "AUDITS_PAGE_HIDE_MESSAGE",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "audits": Array [
          Object {
            "action": "update",
            "action_display_name": "updated",
            "affected_locations": Array [
              Object {
                "name": "test_loc1",
                "url": "/locations/2-test_loc1/edit",
              },
              Object {
                "name": "test_loc2",
                "url": "/locations/6-test_loc2/edit",
              },
              Object {
                "name": "test_loc3",
                "url": "/locations/9-test_loc3/edit",
              },
            ],
            "affected_organizations": Array [
              Object {
                "name": "test_org1",
                "url": "/organizations/1-test_org1/edit",
              },
              Object {
                "name": "test_org2",
                "url": "/organizations/3-test_org2/edit",
              },
              Object {
                "name": "test_org3",
                "url": "/organizations/5-test_org3/edit",
              },
            ],
            "allowed_actions": Array [
              Object {
                "css_class": "btn btn-default",
                "title": "Host details",
                "url": "/hosts/host-foo.example.com",
              },
            ],
            "associated_id": null,
            "associated_name": null,
            "associated_type": null,
            "audit_title": "host-foo.example.com",
            "audit_title_url": "/audits?search=type+%3D+host+and+auditable_id+%3D+9",
            "auditable_id": 9,
            "auditable_name": "host-foo.example.com",
            "auditable_type": "Host::Base",
            "audited_changes": Object {
              "comment": Array [
                "",
                "This is info about host for audit",
              ],
              "root_pass": Array [
                "[redacted]",
                "[redacted]",
              ],
            },
            "audited_changes_with_id_to_label": Array [
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[redacted]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "[redacted]",
                  },
                ],
                "name": "Root pass",
              },
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[empty]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "This is info about host for audit",
                  },
                ],
                "name": "Comment",
              },
            ],
            "audited_type_name": "Host",
            "comment": null,
            "created_at": "2018-08-13 00:34:55 -1100",
            "id": 234,
            "remote_address": "127.0.0.1",
            "request_uuid": "c134239d-8ac3-494b-9962-35133fe153ba",
            "user_id": 4,
            "user_info": Object {
              "audit_path": "/audits?search=id+%3D+234",
              "display_name": "Admin ",
              "login": "admin",
              "search_path": "/audits?search=user+%3D+admin",
            },
            "user_type": null,
            "username": "Admin User",
            "version": 2,
          },
        ],
        "itemCount": 1,
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch empty Audits 1`] = `
Array [
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "audits": Array [],
        "itemCount": 0,
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch nextAudits 1`] = `
Array [
  Array [
    Object {
      "type": "AUDITS_PAGE_NEXT_PENDING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "itemCount": 1,
        "nextAudits": Array [
          Object {
            "action": "update",
            "action_display_name": "updated",
            "affected_locations": Array [
              Object {
                "name": "test_loc1",
                "url": "/locations/2-test_loc1/edit",
              },
              Object {
                "name": "test_loc2",
                "url": "/locations/6-test_loc2/edit",
              },
              Object {
                "name": "test_loc3",
                "url": "/locations/9-test_loc3/edit",
              },
            ],
            "affected_organizations": Array [
              Object {
                "name": "test_org1",
                "url": "/organizations/1-test_org1/edit",
              },
              Object {
                "name": "test_org2",
                "url": "/organizations/3-test_org2/edit",
              },
              Object {
                "name": "test_org3",
                "url": "/organizations/5-test_org3/edit",
              },
            ],
            "allowed_actions": Array [
              Object {
                "css_class": "btn btn-default",
                "title": "Host details",
                "url": "/hosts/host-foo.example.com",
              },
            ],
            "associated_id": null,
            "associated_name": null,
            "associated_type": null,
            "audit_title": "host-foo.example.com",
            "audit_title_url": "/audits?search=type+%3D+host+and+auditable_id+%3D+9",
            "auditable_id": 9,
            "auditable_name": "host-foo.example.com",
            "auditable_type": "Host::Base",
            "audited_changes": Object {
              "comment": Array [
                "",
                "This is info about host for audit",
              ],
              "root_pass": Array [
                "[redacted]",
                "[redacted]",
              ],
            },
            "audited_changes_with_id_to_label": Array [
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[redacted]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "[redacted]",
                  },
                ],
                "name": "Root pass",
              },
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[empty]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "This is info about host for audit",
                  },
                ],
                "name": "Comment",
              },
            ],
            "audited_type_name": "Host",
            "comment": null,
            "created_at": "2018-08-13 00:34:55 -1100",
            "id": 234,
            "remote_address": "127.0.0.1",
            "request_uuid": "c134239d-8ac3-494b-9962-35133fe153ba",
            "user_id": 4,
            "user_info": Object {
              "audit_path": "/audits?search=id+%3D+234",
              "display_name": "Admin ",
              "login": "admin",
              "search_path": "/audits?search=user+%3D+admin",
            },
            "user_type": null,
            "username": "Admin User",
            "version": 2,
          },
        ],
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
  Array [
    Object {
      "type": "AUDITS_PAGE_NEXT_RESOLVED",
    },
  ],
]
`;

exports[`AuditsPage actions should fetch prevAudits 1`] = `
Array [
  Array [
    Object {
      "type": "AUDITS_PAGE_PREV_PENDING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_SHOW_LOADING",
    },
  ],
  Array [
    Object {
      "type": "LAYOUT_HIDE_LOADING",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "itemCount": 1,
        "prevAudits": Array [
          Object {
            "action": "update",
            "action_display_name": "updated",
            "affected_locations": Array [
              Object {
                "name": "test_loc1",
                "url": "/locations/2-test_loc1/edit",
              },
              Object {
                "name": "test_loc2",
                "url": "/locations/6-test_loc2/edit",
              },
              Object {
                "name": "test_loc3",
                "url": "/locations/9-test_loc3/edit",
              },
            ],
            "affected_organizations": Array [
              Object {
                "name": "test_org1",
                "url": "/organizations/1-test_org1/edit",
              },
              Object {
                "name": "test_org2",
                "url": "/organizations/3-test_org2/edit",
              },
              Object {
                "name": "test_org3",
                "url": "/organizations/5-test_org3/edit",
              },
            ],
            "allowed_actions": Array [
              Object {
                "css_class": "btn btn-default",
                "title": "Host details",
                "url": "/hosts/host-foo.example.com",
              },
            ],
            "associated_id": null,
            "associated_name": null,
            "associated_type": null,
            "audit_title": "host-foo.example.com",
            "audit_title_url": "/audits?search=type+%3D+host+and+auditable_id+%3D+9",
            "auditable_id": 9,
            "auditable_name": "host-foo.example.com",
            "auditable_type": "Host::Base",
            "audited_changes": Object {
              "comment": Array [
                "",
                "This is info about host for audit",
              ],
              "root_pass": Array [
                "[redacted]",
                "[redacted]",
              ],
            },
            "audited_changes_with_id_to_label": Array [
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[redacted]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "[redacted]",
                  },
                ],
                "name": "Root pass",
              },
              Object {
                "change": Array [
                  Object {
                    "css_class": "show-old",
                    "id_to_label": "[empty]",
                  },
                  Object {
                    "css_class": "show-new",
                    "id_to_label": "This is info about host for audit",
                  },
                ],
                "name": "Comment",
              },
            ],
            "audited_type_name": "Host",
            "comment": null,
            "created_at": "2018-08-13 00:34:55 -1100",
            "id": 234,
            "remote_address": "127.0.0.1",
            "request_uuid": "c134239d-8ac3-494b-9962-35133fe153ba",
            "user_id": 4,
            "user_info": Object {
              "audit_path": "/audits?search=id+%3D+234",
              "display_name": "Admin ",
              "login": "admin",
              "search_path": "/audits?search=user+%3D+admin",
            },
            "user_type": null,
            "username": "Admin User",
            "version": 2,
          },
        ],
      },
      "type": "AUDITS_PAGE_FETCH",
    },
  ],
  Array [
    Object {
      "type": "AUDITS_PAGE_PREV_RESOLVED",
    },
  ],
]
`;

exports[`AuditsPage actions should fetchAndPush 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "loadingBool": true,
        "page": 1,
        "perPage": 20,
        "searchQuery": "",
      },
      "type": "AUDITS_PAGE_CHANGE_PARAMS",
    },
  ],
]
`;

exports[`AuditsPage actions should fetchingNext 1`] = `
Object {
  "type": "AUDITS_PAGE_NEXT_PENDING",
}
`;

exports[`AuditsPage actions should fetchingPrev 1`] = `
Object {
  "type": "AUDITS_PAGE_PREV_PENDING",
}
`;

exports[`AuditsPage actions should initializeAudits 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "page": 2,
        "perPage": 20,
        "searchQuery": "search",
      },
      "type": "AUDITS_PAGE_CHANGE_PARAMS",
    },
  ],
]
`;

exports[`AuditsPage actions should initializeAudits w/replace 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "page": 2,
        "perPage": 20,
        "searchQuery": "search",
      },
      "type": "AUDITS_PAGE_CHANGE_PARAMS",
    },
  ],
]
`;

exports[`AuditsPage actions should resolvedNext 1`] = `
Object {
  "type": "AUDITS_PAGE_NEXT_RESOLVED",
}
`;

exports[`AuditsPage actions should resolvedPrev 1`] = `
Object {
  "type": "AUDITS_PAGE_PREV_RESOLVED",
}
`;
